app/Hutch.Relay/Controllers/TaskController.cs

```diff
@@ -57,16 +61,29 @@ public async Task<IActionResult> Result(Guid uuid, string collectionId, [FromBod
     }
 
     // Update the SubTask results
-    await relaySubTaskService.SetResult(uuid, JsonSerializer.Serialize(result));
-
+    await relayTaskService.SetSubTaskResult(uuid, JsonSerializer.Serialize(result));
     // Check if there are incomplete Subtasks that belong to the same Task
-    var incompleteSubTasks = await relaySubTaskService.ListIncomplete(subtask.RelayTask.Id);
+    var incompleteSubTasks = await relayTaskService.ListSubTasks(subtask.RelayTask.Id, incompleteOnly: true);
     //  If all Subtasks on the parent Task received - then submit to TaskApi
-    if (!incompleteSubTasks.Any())
+    if (!incompleteSubTasks.ToList().Any())
     {
-      //TODO Aggregate SubTasks result counts
-
-      await taskApiService.SubmitResults(subtask.RelayTask, result);
+      // Get all SubTasks for the Task
+      var subTasks = await relayTaskService.ListSubTasks(subtask.RelayTask.Id, incompleteOnly: false);
+      //Aggregate SubTasks Result.Count
+      var aggregateCount = resultsService.AggregateResults(subTasks.ToList());
+      var finalResult = new JobResult()
+      {
+        Uuid = subtask.RelayTask.Id,
+        CollectionId = apiClientOptions.CollectionId ??
+                       throw new ArgumentException(nameof(apiClientOptions.CollectionId)),
+        Message = result.Message,
+        ProtocolVersion = result.ProtocolVersion,
+        Results = new QueryResult()
+        {
+          Count = aggregateCount,
+        }
+      };
```

Comment:
I think `AggregateResults` should do more of this work? It could defer to multiple methods internally, but I think the interface frm the controller's point of view should be `Give me the final result for this parent task` and it shouldn't have to put in more effort than that.

Then the hosted service will do the same, and we don't have to duplicate the fetching and model building bits

Reply:
Do this refactoring in #93 ? then we can merge